core/shared/src/main/scala/fs2/async/mutable/Queue.scala

```diff
@@ -94,7 +107,7 @@ object Queue {
       * @param queue    Queue, expressed as vector for fast cons/uncons from head/tail
       * @param deq      A list of waiting dequeuers, added to when queue is empty
       */
-    case class State(queue: Vector[A], deq: Vector[Async.Ref[F,A]])
+    case class State(queue: Vector[A], deq: Vector[Either[Async.Ref[F,A], Async.Ref[F,Vector[A]]]])
```

Comment:
What will happen if we will have callback  like Async[Ref[Chunk[A]] ??? Queue may be then Vector[Chunk[A]] instead of just Vector[A], or perhaps just Chunk[A]. Not sure about chunk characteristics in these cases.